spec/controllers/churn_controller_spec.rb

```diff
@@ -0,0 +1,203 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+require "inertia_rails/rspec"
+
+RSpec.describe ChurnController, type: :controller, inertia: true do
+  let(:user) { create(:user) }
+  let!(:product) { create(:subscription_product, user: user) }
+
+  before do
+    sign_in user
+  end
+
+  describe "GET #show" do
+    context "when user has subscription products" do
+      it "renders the Churn/Show component" do
+        get :show
+        expect(response).to be_successful
+        expect(inertia.component).to eq("Churn/Show")
+      end
+
+      it "includes churn_props with subscription products" do
+        get :show
+        expect(response).to be_successful
+        expect(inertia.props[:churn_props]).to include(
+          has_subscription_products: true,
+          products: array_including(
+            hash_including(
+              id: product.id,
+              name: product.name,
+              unique_permalink: product.unique_permalink,
+              alive: true
+            )
+          )
+        )
+      end
+
+      it "does not evaluate lazy props on initial render" do
+        get :show
+        expect(response).to be_successful
+        expect(inertia.props).not_to have_key(:churn_data)
+      end
+
+      it "calls LargeSeller.create_if_warranted" do
+        expect(LargeSeller).to receive(:create_if_warranted).with(user)
+        get :show
+      end
+
+      it "initializes CreatorAnalytics::Churn service with correct parameters" do
+        expect(CreatorAnalytics::Churn).to receive(:new).with(
+          user: user,
+          params: instance_of(ActionController::Parameters)
+        ).and_call_original
+        get :show
+      end
+
+      it "calls service methods for data retrieval" do
+        service_instance = instance_double(CreatorAnalytics::Churn)
+        allow(CreatorAnalytics::Churn).to receive(:new).and_return(service_instance)
+
+        expect(service_instance).to receive(:has_subscription_products?).and_return(true)
+        expect(service_instance).to receive(:available_products).and_return([product])
+        allow(service_instance).to receive(:fetch_churn_data).and_return(nil)
+
+        get :show
+      end
+
+      it "uses InertiaRails.optional for lazy loading churn_data" do
+        expect(InertiaRails).to receive(:optional).and_call_original
+        get :show
+      end
+    end
+
+    context "when user has no subscription products" do
+      let!(:product) { create(:product, user: user) }
+
+      it "renders successfully and indicates no subscription products" do
+        get :show
+        expect(response).to be_successful
+        expect(inertia.props[:churn_props]).to include(
+          has_subscription_products: false
+        )
+      end
+
+      it "returns empty products array" do
+        get :show
+        expect(inertia.props[:churn_props][:products]).to be_empty
+      end
```

Comment:
Group this in one test. I know Rspec wants you to have one "it" statement per test, but we have many of them so we need to find ways to make them faster.

Also, test that the churn_data prop is empty here